.github/tests/lm_tests.py

```diff
@@ -21,35 +21,83 @@ def test_filter_operation(setup_models):
+            "Just finished a great book and feel inspired!",
+            "Had a lovely meal with friends, life is good!",
+            "Everything is going as planned, couldn't be happier.",
+            "Feeling super motivated and ready to take on challenges!",
+            "I appreciate all the small things that bring me joy.",
+
+            # Negative examples
+            "I am very sad.",
+            "Today has been really tough; I feel exhausted.",
+            "I'm feeling pretty down about how things are going.",
+            "I’m overwhelmed with all these challenges.",
+            "It’s hard to stay positive when things keep going wrong.",
+            "I feel so alone and unappreciated.",
+            "My energy is low, and nothing seems to cheer me up.",
+            "Feeling anxious about everything lately.",
+            "I’m disappointed with the way my project turned out.",
+            "Today has been one of those days where everything goes wrong.",
+            "Life feels really overwhelming right now.",
+            "I can't seem to find any motivation these days.",
+            "I’m worried about the future and what it holds.",
+            "It's been a stressful day, and I feel mentally drained.",
+            "I feel like I'm falling behind everyone else.",
+            "Just can't seem to catch a break recently.",
+            "I’m really struggling to keep up with all my responsibilities.",
+            "Had an argument with a close friend, feeling hurt.",
+            "I don’t feel supported by my team at work.",
+            "Life has been tough lately, and I’m feeling down.",
+        ]
+    }
+
     df = pd.DataFrame(data)
     user_instruction = "{Text} is a positive sentiment"
 
     # All filters resolved by the helper model
-    filtered_df, stats = df.sem_filter(user_instruction, cascade_threshold=0, return_stats=True)
-    assert stats["filters_resolved_by_large_model"] == 0, stats
-    assert stats["filters_resolved_by_helper_model"] == 2, stats
-    expected_df = pd.DataFrame({"Text": ["I am really exicted to go to class today!"]})
-    assert filtered_df.equals(expected_df)
-
-    # All filters resolved by the large model
-    filtered_df, stats = df.sem_filter(user_instruction, cascade_threshold=1.01, return_stats=True)
-    assert stats["filters_resolved_by_large_model"] == 2, stats
-    assert stats["filters_resolved_by_helper_model"] == 0, stats
-    assert filtered_df.equals(expected_df)
+    filtered_df, stats = df.sem_filter(
+        user_instruction=user_instruction,
+        learn_cascade_threshold_sample_percentage=0.5,
+        recall_target=0.9,
+        precision_target=0.9,
+        failure_probability=0.2,
+        return_stats=True,
+    )
+
+    assert "I am really excited to go to class today!" in filtered_df["Text"].values
+    assert "I am very sad" not in filtered_df["Text"].values
+    assert stats["filters_resolved_by_helper_model"] > 0, stats
```

Comment:
Just a simple question for my own understanding - is `assert stats["filters_resolved_by_helper_model"] > 0` ever going to be flaky depending on the large/small model. Or is this super safe?

Reply:
Good question -- I think achieving >0 helper calls with a .9/.9 targets for this task should be simple enough for language models that I really wouldn't expect it to be flaky. Worst case if it comes to it the line can be removed, though I'd personally be a bit surprised if it failed.

Reply:
Thanks for clarifying, let's keep it then